src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.Methods.cs

```diff
@@ -123,7 +123,18 @@ public static void WriteMethodBodyStatement(this CodeWriter writer, MethodBodySt
                         WriteMethodBodyStatement(writer, foreachStatement.Body.AsStatement());
                         writer.LineRaw("}");
                     }
-
+                    break;
+                case ForStatement forStatement:
+                    using (writer.AmbientScope())
+                    {
+                        writer.AppendRaw("for (");
+                        writer.Append($"int {forStatement.Indexer:D} = 0; {forStatement.Indexer} < {forStatement.Enumerable}.Length; {forStatement.Indexer}++)");
+                        writer.LineRaw("{");
+                        writer.Append($"{forStatement.ItemType} {forStatement.Item:D} = {forStatement.Enumerable}[{forStatement.Indexer}];");
```

Comment:
I believe this is not a good practice for a new statement.
This statement corresponds to 
```
for (<declOfIndexer>; <conditionOfIndexer>; <steperOfIndexer>)
{
// statements
}
```
I believe we should define this statement as this.
<declOfIndexer> is an "AssignmentExpression" which we do not have right now, we could create one.
<conditionOfIndexer> is a `BoolExpression`
<steperOfIndexer> is just simply a `ValueExpression`, and when construct this, we could use `UnaryExpression` to have `i++`.

And this line:
```
$"{forStatement.ItemType} {forStatement.Item:D} = {forStatement.Enumerable}[{forStatement.Indexer}];"
```
should not show up here, instead if we need this line in your scenario, we should always add this statement to the for statement as its first member

Reply:
Thanks, I've updated based on your guidance.